code/workspaces/web-app/src/components/projects/updateProjectFormValidator.js

```diff
@@ -0,0 +1,25 @@
+import validate from 'validate.js';
+
+const constraints = {
+  name: {
+    presence: true,
+    length: { minimum: 2 },
+  },
+  description: {
+    presence: false,
+  },
+  collaborationLink: {
+    presence: false,
+  },
+};
+
+validate.formatters.reduxForm = errors => errors.reduce(errorReducer, {});
+
+function errorReducer(accumulator, error) {
+  accumulator[error.attribute] = error.error; // eslint-disable-line no-param-reassign
+  return accumulator;
+}
+
+const syncValidate = values => validate(values, constraints, { format: 'reduxForm' });
```

Comment:
Name isn't unique at the moment, only project key, it might make sense to look at that in the end as duplicates would make the list of projects confusing, but doesn't do any worse than that.

Reply:
Ahhh that's the one, I couldn't remember where the uniqueness came from!